sycl/source/queue.cpp

```diff
@@ -45,6 +45,17 @@ queue::queue(const context &syclContext, const device_selector &deviceSelector,
       asyncHandler, detail::getQueueOrder(propList), propList);
 }
 
+#ifndef __SYCL_DISABLE_QUEUE_CONSTRUCTORS_EXTENSION__
```

Comment:
I think this macro makes sense in headers only as library is built separately from user's application.

Reply:
Good point.
But then, do you compile outside of the class declaration, with a different `.hpp` then that would have always? Is it an ABI issue?
Otherwise, is this  a real issue having always this extension?
There is no negative testing in the CTS...

Reply:
Thank you. I removed the macro from queue.cpp